addons/account/models/account_move_line.py

```diff
@@ -1541,9 +1543,20 @@ def write(self, vals):
             if line.parent_state == 'posted' and any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in protected_fields['tax']):
                 line._check_tax_lock_date()
 
-            # Check the reconciliation.
-            if any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in protected_fields['reconciliation']):
-                line._check_reconciliation()
+            # Break the reconciliation.
+            if any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in protected_fields['reconciliation']) and (line.matched_debit_ids or line.matched_credit_ids):
+                lines_to_unreconcile += line
+                partials = line.matched_debit_ids + line.matched_credit_ids
+                st_lines_to_unreconcile += (partials.debit_move_id + partials.credit_move_id).statement_line_id
```

Comment:
```suggestion
                moves = line.matched_debit_ids.debit_move_id + line.matched_credit_ids.credit_move_id
                st_lines_to_unreconcile += moves.statement_line_id
```
If the move line itself is a statement line, the move line could be unlink during the write that will raise a "record doesn't exist".